### ISSUE TITLE
Cells with embedded carriage returns are not surrounded by quotes

### ISSUE THREAD

**Issue by lespea** [uniVocity/univocity-parsers#315]
If you write a string that contains a carriage return I thought that it would be escaped in the CSV.  As it is currently, excel cannot open the generated csv without errors.

#### Test code

```java
import com.univocity.parsers.csv.Csv;
import com.univocity.parsers.csv.CsvWriter;

import java.nio.charset.StandardCharsets;
import java.nio.file.Files;
import java.nio.file.Paths;

public class CsvTest {
    public static void main(String[] args) throws Exception {
        final CsvWriter csv = new CsvWriter(
              Files.newBufferedWriter(Paths.get("blah.csv"), StandardCharsets.UTF_8),
              Csv.writeExcel()
        );

        try {
            csv.writeHeaders("Label", "Value", "Trailing");

            String[][] rows = {
                  {"Start newline", "\nabc123", "Trailing"},
                  {"Middle newline", "abc\n123", "Trailing"},
                  {"End newline", "abc123\n", "Trailing"},
                  {"Start cr", "\rabc123", "Trailing"},
                  {"Middle cr", "abc\r123", "Trailing"},
                  {"End cr", "abc123\r", "Trailing"},
                  {"Start both", "\n\rabc123", "Trailing"},
                  {"Middle both", "abc\n\r123", "Trailing"},
                  {"End both", "abc123\n\r", "Trailing"},
                  };

            for (String[] row : rows) {
                csv.writeRow(row);
            }
        } finally {
            csv.close();
        }
    }
}
```

#### Output
```
bat blah.csv
───────┬──────────────────────────────────────
       │ File: blah.csv
───────┼──────────────────────────────────────
   1   │ Label,Value,Trailing
   2   │ Start newline,"
   3   │ abc123",Trailing
   4   │ Middle newline,"abc
   5   │ 123",Trailing
   6   │ End newline,"abc123
   7   │ ",Trailing
   8   │ Start cr,^Mabc123,Trailing
   9   │ Middle cr,abc^M123,Trailing
  10   │ End cr,abc123,Trailing
  11   │ Start both,"
  12   │ ^Mabc123",Trailing
  13   │ Middle both,"abc
  14   │ ^M123",Trailing
  15   │ End both,"abc123
  16   │ ",Trailing
───────┴──────────────────────────────────────


```

**Comment by jbax** [comment]
Thanks I'll fix this asap.

By default it will only escape the `\r` or `\n` if these have been configured in the CsvFormat. 

```
CsvWriterSettings settings = Csv.writeExcel(); //format is set with line ending = `\r\n`
settings.getFormat().getnormalizedNewline(); // defaults to `\n`
```

The normalizedNewLine is the key here. If found on the value being written, it will be transformed into the line separator defined in the format. A value like `\nabc` will be written to the output as `"\r\nabc"`. It won't do anything to `\rabc` though.

An additional setting is required to make the writer quote the field if it finds this `\r`
```
settings.setQuotationTriggers('\r');
```

Now `\rabc` should be written as `"\r\nabc"` .

If you want to have the field contain the original line ending, and ensure the value is written as `"\rabc"`, you need an additional setting:

```
settings.setNormalizeLineEndingsWithinQuotes(false);
```

However I don't think Excel will like that one.

Hope this helps. CSV is a mixed bag and supporting all variations of intended input/output formats makes the API a bit more complex to use.

**Comment by jbax** [comment]
Done. I've released a 2.8.2-SNAPSHOT build you can use to test this. Thank you for using our parsers!

**Comment by lespea** [comment]
2.8.2 works perfect; thanks for the quick reply!

```
bat blah.csv
───────┬──────────────────────────────────────
       │ File: blah.csv
───────┼──────────────────────────────────────
   1   │ Label,Value,Trailing
   2   │ Start newline,"
   3   │ abc123",Trailing
   4   │ Middle newline,"abc
   5   │ 123",Trailing
   6   │ End newline,"abc123
   7   │ ",Trailing
   8   │ Start cr,"^Mabc123",Trailing
   9   │ Middle cr,"abc^M123",Trailing
  10   │ End cr,"abc123",Trailing
  11   │ Start both,"
  12   │ ^Mabc123",Trailing
  13   │ Middle both,"abc
  14   │ ^M123",Trailing
  15   │ End both,"abc123
  16   │ ",Trailing
───────┴─────────────────────────────────────
```
